### PR TITLE
feat: add c++ api

### PR DESCRIPTION
**Pull request by ddiakiteaneo**
This PR is created since the last one has been accidentally closed. The first PR with the conversation history can be found [here](https://github.com/aneoconsulting/ArmoniK.Api/pull/261).

### CODE REVIEW

File: packages/cpp/ArmoniK.Api.Client/header/submitter/SessionContext.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+   * @param session_id A string representing the session ID.
+   */
+  void set_session_id(const std::string& session_id);
+
+  /**
+   * @brief Gets the task options for the current session.
+   *
+   * @return A reference to the task options object.
+   */
+  auto get_task_options() -> armonik::api::grpc::v1::TaskOptions& { return task_options_; }
```
Comment (lemaitre-aneo) [inline packages/cpp/ArmoniK.Api.Client/header/submitter/SessionContext.h:43]:
```suggestion
  /**
   * @brief Gets the session ID for the current session.
   *
   * @return A reference to the session ID string.
   */
  auto session_id() -> std::string& { return session_id_; }

  /**
   * @brief Gets the task options for the current session.
   *
   * @return A reference to the task options object.
   */
  auto task_options() -> armonik::api::grpc::v1::TaskOptions& { return task_options_; }

```

File: packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+ * @brief Cancel a session.
+ *
+ * @param session_id The session id.
+ */
+bool SubmitterClient::cancel_session(std::string_view session_id)
+{
+  std::shared_ptr<Channel> channel = grpc::CreateChannel(
+      "172.30.39.223:5001",
+                                grpc::InsecureChannelCredentials());
+  auto stub = armonik::api::grpc::v1::sessions::Sessions::NewStub(channel);
```
Comment (lemaitre-aneo) [inline packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp:76]: You should not recreate a new channel here. Plus, the channel shall not be hardcoded.

Maybe you could remove the cancel_session rpc for now, and add it back in a new PR where you will tackle the multiple stub problem.

File: packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+  /**
+   * @brief Submits tasks with dependencies to the session context.
+   * @param session_context The session context.
+   * @param payloads_with_dependencies A vector of tuples containing the payload, its data, and its dependencies.
+   * @param max_retries The maximum number of retries for submitting tasks.
+   * @return A vector of submitted task IDs.
+   */
+  std::tuple<std::vector<std::string>, std::vector<std::string>> submit_tasks_with_dependencies(
+      SessionContext& session_context,
```
Comment (lemaitre-aneo) [inline packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h:106]:
```suggestion
      std::string session_id, armonik::api::grpc::v1::TaskOptions task_options,
```

File: packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+  stub_ = std::move(stub);
+}
+
+/**
+ * @brief Create a new session.
+ * @param partition_ids The partitions ids.
+ * @param default_task_options The default task options.
+ */
+std::string SubmitterClient::create_session(TaskOptions default_task_options,
+                                            const std::vector<std::string>& partition_ids = {"default"})
```
Comment (lemaitre-aneo) [inline packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp:44]:
```suggestion
                                            const std::vector<std::string>& partition_ids = {})
```

File: packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+   * @param task_requests The vector of task requests.
+   * @param session_id The session ID.
+   * @param task_options The task options.
+   * @param chunk_max_size The maximum chunk size.
+   * @return A vector of future large task request objects.
+   */
+  static auto to_request_stream(
+    const std::vector<armonik::api::grpc::v1::TaskRequest>& task_requests,
+    std::string& session_id,
+    const armonik::api::grpc::v1::TaskOptions& task_options,
```
Comment (lemaitre-aneo) [inline packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h:72]:
```suggestion
    std::string session_id,
    armonik::api::grpc::v1::TaskOptions task_options,
```

File: packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h
Comment (lemaitre-aneo) [inline packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h:1]: The interface lacks consistency in the signature of the RPCs.

File: packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+  /**
+   * @brief Creates tasks asynchronously with the specified options and requests.
+   * @param session_id The session ID.
+   * @param task_options The task options.
+   * @param task_requests The vector of task requests.
+   * @return A future create task reply object.
+   */
+  std::future<armonik::api::grpc::v1::submitter::CreateTaskReply> create_tasks_async(
+    std::string& session_id, const armonik::api::grpc::v1::TaskOptions& task_options,
+    const std::vector<armonik::api::grpc::v1::TaskRequest> task_requests);
```
Comment (lemaitre-aneo) [inline packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h:96]:
```suggestion
    std::string session_id, armonik::api::grpc::v1::TaskOptions task_options,
    const std::vector<armonik::api::grpc::v1::TaskRequest>& task_requests);
```

File: packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+  /**
+   * @brief Submits tasks with dependencies to the session context.
+   * @param session_context The session context.
+   * @param payloads_with_dependencies A vector of tuples containing the payload, its data, and its dependencies.
+   * @param max_retries The maximum number of retries for submitting tasks.
+   * @return A vector of submitted task IDs.
+   */
+  std::tuple<std::vector<std::string>, std::vector<std::string>> submit_tasks_with_dependencies(
```
Comment (lemaitre-aneo) [inline packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h:105]: Tuple of vectors? or vector of tuples?

File: packages/cpp/ArmoniK.Api.Client/source/submitter/SessionContext.cpp [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+SessionContext::SessionContext(armonik::api::grpc::v1::TaskOptions task_options) : task_options_(
+    std::move(task_options))
+{
+}
+
+void SessionContext::set_session_id(const std::string& session_id)
+{
+  session_id_ = session_id;
+}
```
Comment (lemaitre-aneo) [inline packages/cpp/ArmoniK.Api.Client/source/submitter/SessionContext.cpp:11]: These should be defined in the header to enable inlining

File: packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp [hunk 9]
```diff
@@ -0,0 +1,452 @@
...
+  CreateSessionReply reply;
+  
+  Status status = stub_->CreateSession(&context_, request, &reply);
+  if (!status.ok())
+  {
+    std::stringstream message;
+    message << "Error: " << status.error_code() << ": " << status.error_message()
+            << ". details : " << status.error_details() << std::endl;
+    std::cout << "CreateSession rpc failed: " << std::endl;
+    throw std::runtime_error(message.str().c_str());
```
Comment (lemaitre-aneo) [inline packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp:61]: You should have a custom exception class for this.

File: packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp [hunk 10]
```diff
@@ -0,0 +1,452 @@
...
+  {
+    CreateLargeTaskRequest_InitRequest create_large_task_request_init;
+    create_large_task_request_init.set_session_id(session_id);
+    *create_large_task_request_init.mutable_task_options() = task_options;
+
+    CreateLargeTaskRequest create_large_task_request;
+    *create_large_task_request.mutable_init_request() = create_large_task_request_init;
+
+    return std::vector{create_large_task_request};
+  }));
```
Comment (lemaitre-aneo) [inline packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp:125]: Once you take `session_id` and `task_options` by value, you can move out those values into the lambda:
```suggestion
  CreateLargeTaskRequest>>> SubmitterClient::to_request_stream(const std::vector<TaskRequest>& task_requests,
                                                                  std::string session_id,
                                                                  TaskOptions task_options,
                                                                  const size_t chunk_max_size)
{
  std::vector<std::future<std::vector<CreateLargeTaskRequest>>> async_chunk_payload_tasks;
  async_chunk_payload_tasks.push_back(std::async([session_id = std::move(session_id), task_options = std::move(task_options)]()
  {
    CreateLargeTaskRequest_InitRequest create_large_task_request_init;
    create_large_task_request_init.set_session_id(std::move(session_id));
    create_large_task_request_init.set_task_options(std::move(task_options));

    CreateLargeTaskRequest create_large_task_request;
    create_large_task_request.set_init_request(std::move(create_large_task_request_init));

    return std::vector{create_large_task_request};
  }));
```

File: packages/cpp/ArmoniK.Api.Common/header/options/ComputePlane.h [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+    {
+      set_worker_address(configuration.get("ComputePlane__WorkerChannel__Address"));
+      set_agent_address(configuration.get(std::string("ComputePlane__AgentChannel__Address")));
+    }
+
+    /**
+     * @brief Returns the server address.
+     * @return A reference to the server address string.
+     */
+    std::string& get_server_address()
```
Comment (lemaitre-aneo) [inline packages/cpp/ArmoniK.Api.Common/header/options/ComputePlane.h:29]:
```suggestion
    std::string_view get_worker_address() const

```

File: packages/cpp/ArmoniK.Api.Common/header/options/ComputePlane.h [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+      if (agent_address.find("unix:") == std::string::npos)
+      {
+        std::stringstream out;
+        out << "unix:" << agent_address;
+        agent_address_ = out.str();
+      }
+      else
+      {
+        agent_address_ = agent_address;
+      }
```
Comment (lemaitre-aneo) [inline packages/cpp/ArmoniK.Api.Common/header/options/ComputePlane.h:67]: This is duplicated. Please factorize this into a private static function.

File: packages/cpp/ArmoniK.Api.Common/header/options/ComputePlane.h [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+      {
+        std::stringstream out;
+        out << "unix:" << socket_address;
+        worker_address_ = out.str();
+      }
+      else
+      {
+        worker_address_ = socket_address;
+      }
+    }
```
Comment (lemaitre-aneo) [inline packages/cpp/ArmoniK.Api.Common/header/options/ComputePlane.h:50]:
```suggestion
    void set_worker_address(std::string socket_address)
    {
      if (socket_address.find("unix:") != 0)
      {
        socket_address.insert(0, "unix:");
      }
      worker_address_ = std::move(socket_address);
    }
```

File: packages/cpp/ArmoniK.Api.Client/header/submitter/SessionContext.h (hunk 1 above)
Comment (ddiakiteaneo) [reply packages/cpp/ArmoniK.Api.Client/header/submitter/SessionContext.h]: Done !

File: packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h (hunk 3 above)
Comment (ddiakiteaneo) [reply packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h]: Noted !
Also, I completely removed the session context class.

File: packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp (hunk 4 above)
Comment (ddiakiteaneo) [reply packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp]: Done !

File: packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h (hunk 6 above)
Comment (ddiakiteaneo) [reply packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h]: Done !

File: packages/cpp/ArmoniK.Api.Client/source/submitter/SessionContext.cpp (hunk 8 above)
Comment (ddiakiteaneo) [reply packages/cpp/ArmoniK.Api.Client/source/submitter/SessionContext.cpp]: I removed the class

File: .docs/content/1.guide/5.cpp.md [hunk 14]
```diff
@@ -0,0 +1,124 @@
+# I. Compilation of ArmoniK.Api.cpp Client and Server on Linux using Docker
```
Comment (esoubiran-aneo) [inline .docs/content/1.guide/5.cpp.md:1]:
```suggestion
# Compilation of ArmoniK.Api.cpp Client and Server on Linux using Docker

```

File: .docs/content/1.guide/5.cpp.md [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+- Make sure you have all the prerequisites installed correctly.
+- Check that you are running PowerShell as an administrator.
+
+## Conclusion
+
+Compiling the Armonik API client and server on Windows can be a complex process.
+
+By following the steps outlined in this guide, you should be able to compile the project successfully and start using the Armonik API on Windows.
+
+# II. Compilation of the Worker ArmoniK.Api.cpp Image for Deployment in Armonik Infrastructure
```
Comment (esoubiran-aneo) [inline .docs/content/1.guide/5.cpp.md:85]: No more than one h1 in a page and you can remove the list effect (II. )

**Review by esoubiran-aneo — APPROVED** [review]

**Review by esoubiran-aneo — CHANGES_REQUESTED** [review]
see comments

File: .docs/content/1.guide/5.cpp.md (hunk 15 above)
Comment (ddiakiteaneo) [reply .docs/content/1.guide/5.cpp.md:85]: What do you mean by h1 ?

**Comment by dbrasseur-aneo** [comment]
https://github.com/aneoconsulting/ArmoniK.Api/actions/runs/5324854576/jobs/9644749802 This needs to be fixed prior to merging

File: packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp [hunk 16]
```diff
@@ -0,0 +1,337 @@
...
+ *
+ * @param session_id The session ID.
+ * @param task_options The task options
+ * @param payloads_with_dependencies A vector of tuples containing task payload
+ * and its dependencies.
+ * @param max_retries Maximum number of retries.
+ * @return A vector of task IDs.
+ */
+std::tuple<std::vector<std::string>, std::vector<std::string>>
+SubmitterClient::submit_tasks_with_dependencies(std::string session_id, TaskOptions task_options,
```
Comment (dbrasseur-aneo) [inline packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp:233]: So only 1 result per task, should be ok for now but it's more ExtC# like than Api like imo. Definitely something to add to TODO

File: packages/cpp/ArmoniK.Api.Common/header/utils/GuuId.h [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+class GuuId {
+public:
+  /**
+   * @brief Generates a random UUID string.
+   *
+   * This method generates a random UUID string, following the format "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx".
+   *
+   * @return A std::string containing the generated UUID.
+   */
+  static std::string generate_uuid() {
```
Comment (dbrasseur-aneo) [inline packages/cpp/ArmoniK.Api.Common/header/utils/GuuId.h:29]: Good enough for now, but it's not the actual algorithm https://datatracker.ietf.org/doc/html/rfc4122#section-4.4

File: packages/cpp/tools/patch/0001-Fix-issue-with-Visual-Studio-2022-toolset.patch
Comment (dbrasseur-aneo) [inline packages/cpp/tools/patch/0001-Fix-issue-with-Visual-Studio-2022-toolset.patch:1]: Should this file be there ?

**Review by dbrasseur-aneo — CHANGES_REQUESTED** [review]

File: packages/cpp/tools/Dockerfile.ubuntu [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+    autoconf \
+    automake \
+    libtool \
+    curl \
+    make \
+    unzip
+
+# Install build-essential and wget, then clean up apt cache to reduce image size
+RUN apt-get update && \
+  apt-get install build-essential wget cmake -y
```
[the remark this answers is not in the document]
Comment (dbrasseur-aneo) [reply packages/cpp/tools/Dockerfile.ubuntu:20]: To fix this kind of issue : https://github.com/aneoconsulting/ArmoniK.Api/actions/runs/5324854576/jobs/9644749802

[the remark this answers is not in the document]
Comment (ddiakiteaneo) [reply packages/cpp/tools/Dockerfile.ubuntu]: Done, thanks!

File: packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp (hunk 16 above)
Comment (ddiakiteaneo) [reply packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp:233]: Yes will do !

File: packages/cpp/ArmoniK.Api.Common/header/utils/GuuId.h (hunk 17 above)
Comment (ddiakiteaneo) [reply packages/cpp/ArmoniK.Api.Common/header/utils/GuuId.h:29]: I agree, it's only used in tests for now and will be removed after.

File: packages/cpp/tools/patch/0001-Fix-issue-with-Visual-Studio-2022-toolset.patch
Comment (ddiakiteaneo) [reply packages/cpp/tools/patch/0001-Fix-issue-with-Visual-Studio-2022-toolset.patch:1]: Removed!

File: .docs/content/1.guide/5.cpp.md [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+
+1. Install Docker on your Linux system. Follow the instructions on the [official Docker documentation](https://docs.docker.com/engine/install/).
+
+2. Clone the repository containing the source code and the necessary scripts.
+
+### Compilation Steps for Linux
+
+1. Open a terminal in the root directory of the cloned repository.
+
+2. Run the `compile.sh` script:
```
Comment (esoubiran-aneo) [inline .docs/content/1.guide/5.cpp.md:17]: Could you add more info about the location of this file. If I strictly follow your guide, I expect that `./compile` is a file at the root of the project. That's not the case.

File: .docs/content/1.guide/5.cpp.md [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+
+```bash
+./compile.sh
+```
+
+The compile.sh script does the following:
```
Comment (esoubiran-aneo) [inline .docs/content/1.guide/5.cpp.md:23]:
```suggestion
The `compile.sh` script does the following:
```

File: .docs/content/1.guide/5.cpp.md [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+
+- Sets the image tag for the Docker image.
+
```
Comment (esoubiran-aneo) [inline .docs/content/1.guide/5.cpp.md:26]:
```suggestion
```

Comment (esoubiran-aneo) [reply .docs/content/1.guide/5.cpp.md:26]: You don't need to add a line between point in a list.

File: .docs/content/1.guide/5.cpp.md [hunk 22]
```diff
@@ -0,0 +1,126 @@
...
+- Determines the absolute paths of the necessary directories (working, proto, build, and install directories).
+
```
[the remark this answers is not in the document]
Comment (esoubiran-aneo) [reply .docs/content/1.guide/5.cpp.md:28]: See https://www.markdownguide.org/basic-syntax/#unordered-lists

File: .docs/content/1.guide/5.cpp.md [hunk 23]
```diff
@@ -0,0 +1,126 @@
...
+- Checks if the Docker image exists. If not, it builds the Docker image using the Dockerfile.ubuntu file.
+
+- Compiles the project source using the Docker image.
```
Comment (esoubiran-aneo) [inline .docs/content/1.guide/5.cpp.md:31]:
```suggestion
- Compiles the project source using the Docker image.

```

Comment (esoubiran-aneo) [reply .docs/content/1.guide/5.cpp.md:31]: You need to a blank line after a list.

File: .docs/content/1.guide/5.cpp.md [hunk 24]
```diff
@@ -0,0 +1,126 @@
...
+
+Now you have successfully compiled the client and server on Linux using Docker.
+
+### Compiling the Client and Server on Windows
+
+This guide explains how to compile the Armonik API client and server on Windows
+
+### Prerequisites Windows
+
+Before getting started, make sure you have the following tools and packages installed on your machine:
```
Comment (esoubiran-aneo) [inline .docs/content/1.guide/5.cpp.md:42]:
```suggestion
Before getting started, make sure you have the following tools and packages installed on your machine:

```

Add a line before your list.

File: .docs/content/1.guide/5.cpp.md [hunk 25]
```diff
@@ -0,0 +1,126 @@
...
+- Chocolatey package manager
+- Grpc 1.54.0 built from source
+- CMake
+- NASM
+
+### Compilation Steps for windows
+
+Follow these steps to compile the Armonik API client and server:
+
+- From a powershell, go to the folder package/cpp/tools
```
Comment (esoubiran-aneo) [inline .docs/content/1.guide/5.cpp.md:58]:
```suggestion
From a PowerShell, go to the folder package/cpp/tools
```

File: .docs/content/1.guide/5.cpp.md [hunk 26]
```diff
@@ -0,0 +1,126 @@
...
+
+### Conclusion
+
+Compiling the Armonik API client and server on Windows can be a complex process.
+
+By following the steps outlined in this guide, you should be able to compile the project successfully and start using the Armonik API on Windows.
+
+## Compilation of the Worker ArmoniK.Api.cpp Image for Deployment in Armonik Infrastructure
+
+### Compilation of the Worker Image for Deployment in Armonik Infrastructure
```
Comment (esoubiran-aneo) [inline .docs/content/1.guide/5.cpp.md:89]: This title is not necessary as it add no information.
```suggestion
```

File: .docs/content/1.guide/5.cpp.md [hunk 27]
```diff
@@ -0,0 +1,126 @@
...
+
+The worker image is a Docker image that is built specifically to be deployed in the Armonik infrastructure. This image contains the necessary dependencies and configurations for the worker to function correctly.
+
+### Prerequisites
+
+1. Install Docker on your Linux system. Follow the instructions on the [official Docker documentation](https://docs.docker.com/engine/install/).
```
Comment (esoubiran-aneo) [inline .docs/content/1.guide/5.cpp.md:95]: You must not add a blank line between item in your list.

File: .docs/content/1.guide/5.cpp.md [hunk 28]
```diff
@@ -0,0 +1,126 @@
...
+
```
Comment (esoubiran-aneo) [inline .docs/content/1.guide/5.cpp.md:96]:
```suggestion
```

File: .docs/content/1.guide/5.cpp.md [hunk 29]
```diff
@@ -0,0 +1,126 @@
...
+2. Clone the repository containing the source code and the necessary scripts.
+
+### Compilation Steps
+
+1. Open a terminal in the root directory of the cloned repository.
+
+2. Run the `build-worker.sh` script:
```
Comment (esoubiran-aneo) [inline .docs/content/1.guide/5.cpp.md:103]: As mentioned at the beginning, I don't understand where this file is.

File: .docs/content/1.guide/5.cpp.md [hunk 30]
```diff
@@ -0,0 +1,126 @@
...
+### Prerequisites
+
+1. Install Docker on your Linux system. Follow the instructions on the [official Docker documentation](https://docs.docker.com/engine/install/).
+
+2. Clone the repository containing the source code and the necessary scripts.
+
+### Compilation Steps
+
+1. Open a terminal in the root directory of the cloned repository.
+
```
Comment (esoubiran-aneo) [inline .docs/content/1.guide/5.cpp.md:102]:
```suggestion
```
You do not need to add a blank line between item in your list.

**Review by esoubiran-aneo — CHANGES_REQUESTED** [review]

File: packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h
Comment (lemaitre-aneo) [reply packages/cpp/ArmoniK.Api.Client/header/submitter/SubmitterClient.h:1]: Work for another PR

File: packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp (hunk 9 above)
Comment (lemaitre-aneo) [reply packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp:53]: Work for another PR

File: packages/cpp/ArmoniK.Api.Common/header/utils/GuuId.h (hunk 17 above)
Comment (lemaitre-aneo) [reply packages/cpp/ArmoniK.Api.Common/header/utils/GuuId.h:29]: Work for another PR

**Review by lemaitre-aneo — APPROVED** [review]

**Review by esoubiran-aneo — APPROVED** [review]

**Review by ngruelaneo — APPROVED** [review]

**Pull request merged**